rides/schema.py

```diff
@@ -40,13 +40,18 @@ class Meta:
 class Query(graphene.ObjectType):
     all_cities = graphene.List(CityType)
     available_rides = graphene.List(RideType)
-    search_by_cities = graphene.List(RideType, start_city_id = graphene.Int(), end_city_id = graphene.Int())
+    search_by_cities = graphene.List(RideType, start_city_id = graphene.Int(), end_city_id = graphene.Int(), departure_time = graphene.types.datetime.Date())
 
     def resolve_all_cities(self, info, **kwargs):
         return City.objects.all()
 
     def resolve_available_rides(self, info, **kwargs):
         return Ride.objects.filter(status='available')
 
-    def resolve_search_by_cities(self, info, start_city_id, end_city_id):
-        return Ride.objects.filter(status='available', start_city_id = start_city_id, end_city_id = end_city_id)
+    def resolve_search_by_cities(self, info, start_city_id, end_city_id, departure_time = None):
+
+        if departure_time and Ride.objects.filter(status='available', start_city_id = start_city_id, end_city_id = end_city_id, departure_time__gte = departure_time):
```

Comment:
if someone gives a departure_time AND there are rides from city x to city y that leave on that day or later, then it returns those rides. 

Otherwise, it just returns all rides from city X to city Y and orders them by depature_time ascending

Reply:
Nice job, I like how dynamic it is